services/QuillLMS/app/controllers/teachers/classroom_manager_controller.rb

```diff
@@ -46,6 +46,14 @@ def retrieve_classrooms_i_teach_for_custom_assigning_activities
     render json: classroom_with_students_json(current_user.classrooms_i_teach)
   end
 
+  def classrooms_and_classroom_units_for_activity_share
+    unit_id = params["unit_id"]
+    render json: {
+      classrooms: classroom_with_students_json(current_user.classrooms_i_teach),
+      classroom_units: unit_id ? ClassroomUnit.where(unit_id: unit_id) : []
```

Comment:
So the `ClassroomUnit` database definition doesn't allow `unit_id` to be `nil`.  This means that no records can be found with that value.  If you call `ClassroomUnit.where(unit_id: nil)` you'll actually get an empty array back.  So you don't need the ternary here.  You can just make this line:`classroom_units: ClassroomUnit.where(unit_id: unit_id)`

Reply:
nice! I'll update